sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_io/base.py

```diff
@@ -126,6 +126,16 @@ def mode(self, mode):
     def description(self) -> str:
         return self._description
 
+    @description.setter
+    def description(self, description):
+        # For un-configured input/output, we build a default data entry for them.
+        self._build_default_data()
+        self._description = description
+        if isinstance(self._data, (Input, Output)):
```

Comment:
If self._data.description value will always be set to description, does this line is needed?

Reply:
I think this change is from feature: "support setting mode for input/output". 
For data which is Input/Output, we set description directly because Input/Output has no private property for type/mode....  
For data which is not Input/Output (actually it will only be InputOutputBase and its subclass), we can't set description directly as it will call setter func with infinite loop